File: Backend/manga-world/src/main/java/com/amakakeru/mangaworld/entity/AuthorPicture.java
package com.amakakeru.mangaworld.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.io.Serializable;

@Data
@Entity
@Table(name = "author_picture")
public class AuthorPicture implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ap_id")
    private Long apId;

    @Lob
    @Column(name = "ap_picture", nullable = false)
    private byte[] apPicture;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "a_id", nullable = false, unique = true)
    private Author author;
}

File: Frontend/manga-world/src/Model/AuthorPicture.ts
import Author from "./Author";

class AuthorPicture {
    apid: number;
    appicture: string;
    author: Author;

    constructor(apid: number, appicture: string, author: Author) {
        this.apid = apid;
        this.appicture = appicture;
        this.author = author;
    }
}

export default AuthorPicture;